server/service/service_osquery.go

```diff
@@ -799,10 +805,15 @@ func (svc *Service) SubmitDistributedQueryResults(
 				}()
 			}
 		}
+		// TODO(mna): is it an issue for the webhook if the policy executions are
+		// temporarily recorded in redis? Obviously it wouldn't see the new flipped
+		// policies on the next run, but would it typically run before the redis
+		// async data is collected and persisted in mysql? Should
+		// FlippingPoliciesForHost take pending redis data into consideration, or
+		// maybe we should impose restrictions between async collection interval
+		// and policy update interval?
```

Comment:
This is the interval configuration-related issue that I mentioned in the backend meeting, if the async collector interval is strictly smaller than the policy update interval, it's probably fine to leave the code as-is (the webhook triggering would be based on whatever is in mysql currently, and on the next run, whatever was in redis temporarily will have been persisted to mysql and everything would work just fine). However, if multiple policy reporting can happen for a host before an async collection is done, then the webhook check might miss some policy result transitions (i.e. it might've gone from pass to fail to pass again, but all that happened while the data was in redis, never in mysql).

Of course, an async collection could also fail even if the intervals were otherwise correct - if we want to be totally sure to trigger the webhooks with no misses, I think the webhook check would need to be aware of the transitive redis data.